tez-runtime-library/src/main/java/org/apache/tez/runtime/library/utils/CodecUtils.java

```diff
@@ -78,7 +78,8 @@ public static InputStream getDecompressedInputStreamWithBufferSize(CompressionCo
       throws IOException {
     String bufferSizeProp = TezRuntimeUtils.getBufferSizeProperty(codec);
     Configurable configurableCodec = (Configurable) codec;
-    int originalSize = configurableCodec.getConf().getInt(bufferSizeProp, DEFAULT_BUFFER_SIZE);
+    int originalSize = bufferSizeProp == null ? DEFAULT_BUFFER_SIZE :
```

Comment:
thanks @jshmchenxi, this looks like an important bugfix
before merging this, could you please clarify what kind of coded have you used?
if bufferSizeProp is null, it means that the codec is not supported by [TezRuntimeUtils.getBufferSizeProperty](https://github.com/apache/tez/blob/master/tez-runtime-library/src/main/java/org/apache/tez/runtime/library/common/TezRuntimeUtils.java#L269)
if it's a codec supported by hadoop, we should add it, otherwise the DEFAULT_BUFFER_SIZE fallback makes sense

Reply:
@abstractdog  thanks for the quick reply! I've updated the patch. In our environment the codec is GzipCodec

Reply:
thanks for the fix, I'm about to merge this soon if no objections
@jteagles: could you please contributor rights to @jshmchenxi to the Tez project in Jira?

Reply:
@abstractdog, I have added contributor rights to @jshmchenxi to the Tez project in jira. I have also added you (@abstractdog) to the administrator role to the Tez which grants permissions to add contributors and other project level changes.